src/prefect/infrastructure/docker.py

```diff
@@ -162,18 +169,39 @@ class DockerContainer(Infrastructure):
     necessary and the API is connectable while bound to localhost.
     """
 
-    image: str = Field(default_factory=get_prefect_image_name)
-    image_pull_policy: ImagePullPolicy = None
+    type: Literal["docker-container"] = Field(
+        "docker-container", description="The type of infrastructure"
+    )
+    image: str = Field(
+        description="Tag of a Docker image to use. Defaults to the Prefect image.",
```

Comment:
When do these need `...` vs just skipping?

Reply:
The `...` let's [Pydantic know that a field is required](https://pydantic-docs.helpmanual.io/usage/models/#required-fields). This field has a `default_factory`, so it isn't required.